toolkit/tools/imagecustomizerapi/partitionsize.go

```diff
@@ -58,3 +59,48 @@ func (s *PartitionSize) UnmarshalYAML(value *yaml.Node) error {
 	}
 	return nil
 }
+
+func (s PartitionSize) MarshalJSON() ([]byte, error) {
+	switch s.Type {
+	case PartitionSizeTypeGrow:
+		return json.Marshal("grow")
+	case PartitionSizeTypeExplicit:
+		return json.Marshal(s.Size.String())
+	default:
+		return json.Marshal(nil)
+	}
+}
+
+func (s *PartitionSize) UnmarshalJSON(data []byte) error {
+	var err error
+
+	var stringValue string
+	err = json.Unmarshal(data, &stringValue)
+	if err != nil {
+		return fmt.Errorf("invalid partition size format: %w", err)
+	}
+
+	switch stringValue {
```

Comment:
Move duplicate code between `UnmarshalYAML` and `UnmarshalJSON` into helper function.

Reply:
done for both files